packages/ui/src/helpers/authenticator/utils.ts

```diff
@@ -33,7 +36,19 @@ export const censorPhoneNumber = (val: string): string => {
   return split.join('');
 };
 
-export const defaultAuthHubHandler: HubHandler = (data, service) => {
+const waitForAutoSignInState = async (service: AuthInterpreter) => {
+  // https://xstate.js.org/docs/guides/interpretation.html#waitfor
+  try {
+    await waitFor(service, (state) =>
+      getActorState(state).matches('autoSignIn')
+    );
+  } catch (e) {
+    // do not throw and ignore the hub event if 'autoSignIn' event was called
+    // in unrelated state.
```

Comment:
Is this getting called extraneously?

Reply:
The `waitForAutoSignInState` is only being called by the `defaultAuthHubHandler`. The `defaultAuthHubHandler` is being called extraneously from the e2e and spec file tests.

Reply:
Will that happen in use cases not pertaining to our e2e tests? Just thinking that this could be really confusing for someone touching this code in the future

Reply:
Talked offline, this does happen with normal usage because Amplify JS can send duplicate hub events. Adding comments to better explain this.

Reply:
a81f953f332ae2726be5c30da16b382a07df3432